### ISSUE TITLE
Feature Request: "Markdown Mode"

### ISSUE THREAD

**Issue by ShadowJonathan** [matrix-org/matrix-wysiwyg#253]
Seeing how WYSIWYG is effectively turning into a Message Composer Driver for Element clients (at this time), I wonder if it's possible to support markdown as a composition method, as an alternate mode to rich text.

The kind of presentation (and/or hinting) I'm thinking about would be something like the following, where text would be stylised according to the effects that the markdown syntax would have on that particular piece of text. This specific image comes from Discord, which recently implemented this in their desktop/web composer:

![image](https://user-images.githubusercontent.com/22740616/199215799-7dd643f9-83cd-4fa5-8fb6-47d288c9fc6f.png)

As you can see, this could enrich the typing experience of people who are primarily used to markdown, and may not wholly work well with only-rich-text.

I think matrix-WYSIWYG is uniquely positioned to allow this composition method to flourish in matrix, if it were to offer this input mode alongside rich text.

**Comment by andybalaam** [comment]
I agree that this style is a nice way of making markdown are really good experience.

At the moment, our team is focussed on making rich editing the best possible experience, and offering a very plain markdown experience for those who want to use it.

So this feature seems like a nice idea, but is not something we are going to devote resources to at the moment.
